lib/ramble/ramble/language/language_base.py

```diff
@@ -63,65 +63,39 @@ def __new__(cls, name, bases, attr_dict):
         # commands:
         # 1. in the order they were defined
         # 2. following the MRO
-
-        attr_dict['_directives_to_be_executed'] = []
-
-        meta_stack = []
-        meta_stack.extend(cls.__bases__)
-        meta_list = []
-        while meta_stack:
-            cur_meta = meta_stack.pop(0)
-
-            meta_stack.extend(cur_meta.__bases__)
-
-            if hasattr(cur_meta, '_directives_to_be_executed'):
-                meta_list.append(cur_meta)
-
-        for meta in meta_list:
-            try:
-                directives = meta._directives_to_be_executed
-                attr_dict['_directives_to_be_executed'].extend(directives)
-                meta._directives_to_be_executed = []
-            except AttributeError:
-                pass
-
-            for directive in meta._directive_names:
-                if directive not in cls._directive_names:
-                    cls._directive_names |= set((directive,))
-
+        attr_dict["_directives_to_be_executed"] = []
         for base in reversed(bases):
             try:
                 directive_from_base = base._directives_to_be_executed
-                attr_dict['_directives_to_be_executed'].extend(
-                    directive_from_base)
+                attr_dict["_directives_to_be_executed"].extend(directive_from_base)
             except AttributeError:
                 # The base class didn't have the required attribute.
                 # Continue searching
                 pass
 
         # De-duplicates directives from base classes
-        attr_dict['_directives_to_be_executed'] = [
-            x for x in llnl.util.lang.dedupe(
-                attr_dict['_directives_to_be_executed'])]
+        attr_dict["_directives_to_be_executed"] = [
```

Comment:
tiny nit: why are you swapping out ' for " ?

Reply:
No real reason. I'm just copying spack's directive meta class, so we stay aligned: https://github.com/spack/spack/blob/develop/lib/spack/spack/directives.py#L148

Reply:
OK! I wonder if we should decide what style we want to follow..